FRAuth/FRAuth/Callbacks/DeviceBindingCallback.swift

```diff
@@ -0,0 +1,221 @@
+            throw AuthError.invalidCallbackResponse(String(describing: json))
+        }
+        
+        for output in outputs {
+            if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.userId, let outputValue = output[CBConstants.value] as? String {
+                userId = outputValue
+            }
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.userName.lowercased(), let outputValue = output[CBConstants.value] as? String {
+                userName = outputValue
+            }
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.authenticationType, let outputValue = output[CBConstants.value] as? String, let outputEnumValue = DeviceBindingAuthenticationType(rawValue: outputValue) {
+                deviceBindingAuthenticationType = outputEnumValue
+            }
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.challenge, let outputValue = output[CBConstants.value] as? String {
+                challenge = outputValue
+            }
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.title, let outputValue = output[CBConstants.value] as? String {
+                title = outputValue
+            }
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.subtitle, let outputValue = output[CBConstants.value] as? String {
+                subtitle = outputValue
+            }
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.description, let outputValue = output[CBConstants.value] as? String {
+                promptDescription = outputValue
+            }
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.timeout, let outputValue = output[CBConstants.value] as? Int {
+                timeout = outputValue
+            }
+        }
+        
+        guard userId != nil, userName != nil, challenge != nil, deviceBindingAuthenticationType != nil, title != nil, subtitle != nil, promptDescription != nil else {
+            throw AuthError.invalidCallbackResponse("Missing an output value")
+        }
+        
+        for input in inputs {
+            if let name = input[CBConstants.name] as? String, name.contains(CBConstants.jws) {
+                self.jwsKey = name
+            }
+            else if let name = input[CBConstants.name] as? String, name.contains(CBConstants.deviceName) {
+                self.deviceNameKey = name
+            }
+            else if let name = input[CBConstants.name] as? String, name.contains(CBConstants.deviceId) {
+                self.deviceIdKey = name
+            }
+            else if let name = input[CBConstants.name] as? String, name.contains(CBConstants.clientError) {
+                self.clientErrorKey = name
```

Comment:
For consistency, do we need self.jwsKey = name or just jwsKey = name, if the global variable and local variable same , people prefer self.